### PR TITLE
upload widget

### PR DESCRIPTION
**Pull request by fritzschoff**

### CODE REVIEW

File: libs/ui/src/lib/upload/widget/upload-widget.component.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class UploadWidgetComponent {
+
+  items$: Observable<UploadState[]> = this.mediaQuery.selectAll();
+
+  constructor(private mediaService: MediaService, private mediaQuery: MediaQuery) { }
+
+  delegateAction(fileName: string, action: 'pause' | 'resume' | 'cancel' | 'clear') {
+    this.mediaService[action](fileName);
```
Comment (pldespaigne) [inline libs/ui/src/lib/upload/widget/upload-widget.component.ts:25]: really cool 👍

File: libs/ui/src/lib/upload/widget/upload-widget.component.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
+<button mat-icon-button (click)="close()" class="close-btn">
+  <mat-icon svgIcon="cross_circle"></mat-icon>
+</button>
+<mat-accordion *ngIf="(items$ | async).length as itemCount" class="mat-card">
```
Comment (pldespaigne) [inline libs/ui/src/lib/upload/widget/upload-widget.component.html:4]: should it not be `(items$ | async).length as itemCount > 0` ?

File: libs/ui/src/lib/upload/widget/upload-widget.component.html [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  <mat-expansion-panel expanded>
+    <mat-expansion-panel-header i18n>Uploading {{ itemCount }} item{ itemCount,
+      plural, =1 {} other {s} }
+    </mat-expansion-panel-header>
```
Comment (pldespaigne) [inline libs/ui/src/lib/upload/widget/upload-widget.component.html:8]: I might be wrong but I think you forgot one pair of `{}`

File: apps/festival/festival/src/app/dashboard/home/home.component.ts [hunk 4]
```diff
@@ -15,13 +17,20 @@ export class HomeComponent implements OnInit, OnDestroy {
...
-  ) {}
+    private orgQuery: OrganizationQuery,
+    private mediaService: MediaService
+  ) { }
+
+  @ViewChild('file') ctrl: ElementRef<HTMLInputElement>;
+  dev() {
+    const file = this.ctrl.nativeElement.files
+    this.mediaService.uploadFile(`dev-max/${Math.random().toString()}`, file)
+  }
```
Comment (pldespaigne) [inline apps/festival/festival/src/app/dashboard/home/home.component.ts:33]: please at least add a `// TODO DO NOT MERGE !!!!` ^^

File: libs/ui/src/lib/upload/widget/upload-widget.component.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+        return '/assets/images/dark/image.webp';
+      case 'pdf':
+        return '/assets/images/dark/pdf.webp';
+      default:
+        return '/assets/images/dark/image.webp';
+    }
+  }
+
+  close() {
+    this.mediaService.detachWidget()
```
Comment (pldespaigne) [inline libs/ui/src/lib/upload/widget/upload-widget.component.ts:45]: you should also call `this.mediaService.clear()` to automatically remove succeeded/canceled uploads

File: libs/utils/src/lib/media/media.service.ts [hunk 6]
```diff
@@ -96,7 +121,23 @@ export class MediaService {
   }
 
   /** Remove every `succeeded` and `canceled` upload */
-  clear() {
-    this.store.remove(upload => isUploading(upload));
+  clear(fileName?: string) {
+    if (fileName) {
+      this.store.remove(upload => upload.id === fileName)
```
Comment (pldespaigne) [inline libs/utils/src/lib/media/media.service.ts:126]: you can simply do `this.store.remove(fileName)`

File: libs/ui/src/lib/upload/widget/upload-widget.component.html (hunk 2 above)
Comment (fritzschoff) [reply libs/ui/src/lib/upload/widget/upload-widget.component.html:4]: 0 is falsy, so no need for checking against a 0

File: libs/ui/src/lib/upload/widget/upload-widget.component.html (hunk 3 above)
Comment (fritzschoff) [reply libs/ui/src/lib/upload/widget/upload-widget.component.html:8]: no the plural pipe has this shit syntax

File: libs/utils/src/lib/media/media.service.ts [hunk 7]
```diff
@@ -1,43 +1,68 @@
...
+    } else {
+      const promises = [];
+      for (let index = 0; index < file.length; index++) {
+        promises.push(this.upload(path, file.item(index), file.item(index).name))
+      }
+      return Promise.all(promises);
+    }
   }
 
   private async upload(path: string, fileOrBlob: Blob | File, fileName: string) {
```
[the remark this answers is not in the document]
Comment (pldespaigne) [reply libs/utils/src/lib/media/media.service.ts:68]: bump

**Pull request merged**